website/source/docs/auth/aws.html.md

```diff
@@ -0,0 +1,1006 @@
+by AWS (public certificate varies by region).
+
+During the login, to establish authenticity of the information provided by the
+client (EC2 instance), the PKCS#7 signature is validated by the backend. Before
+succeeding the login attempt and returning a Vault token, AWS API DescribeInstanceStatus
+is invoked to check if the instance is healthy.
+
+## Authorization workflow
+
+The AMIs that are used by instances should be associated with Vault policies at
+priori, which provides access control primitives on the resources. A successful
+login returns a token. The policies of this token are the same policies that are
+associated with the registered AMI. If `role_tag` option (refer API section) is
+enabled on the AMI, then the policies of the token will be the subset of the
+policies that are associated with the AMI.
+
+## Client Nonce
+
+If an unintended party gets access to the PKCS#7 signature of a particular
+instance, it can impersonate that instance and fetch a Vault token. The design
+of this backend addresses this problem by sharing the responsibility with the
+clients of this backend. The backend will **NOT** be able to distinguish the
+genuineness of the request, during the first login. But once an instance performs
+a successful login, the backend can then thwart the replay-login attempts from
+unintended parties, using a unique nonce that is supplied by the client, during
+its first successful login. The login from an unintended party is detected when
+the instance tries to login for the first time and it fails. A security alert
+should be triggered in such cases.
+
+The client should ensure that it generates unique nonces and makes sure that
+it uses the same nonce for each login attempt. During the first login, the
+backend caches the client nonce in a `whitelist`. For the subsequent login
+requests to succeed, the presented client nonce should match the cached nonce.
+Hence, if the nonce is lost/changed then a token cannot be refreshed (rotated).
+
+## Advanced options and caveats
+
+### Dynamic management of policies via role tags
+If the instance is required to have customized set of policies based on the
+role it plays, it can be achieved by setting `role_tag` option (refer API
+section) on the registered AMI. When this option is set, during the login,
+along with verification of PKCS#7 signature and instance health, the backend
+will query for a specific tag that is attached to the instance. This tag will
+hold information that represents a subset of capabilities that are set on the
+AMI. Hence, a successful login when `role_tag` is enabled on AMI, returns a
+token with the capabilities that are a subset of the capabilities configured
+on the AMI. A `role_tag` can be created using `auth/aws/image/<ami_id>/roletag`
+endpoint and is immutable. The information present in the tag is SHA256 hashed
+and HMAC protected. The key to HMAC is only maintained in the backend.
+
```

Comment:
So there's no way to even list the role tags?  I'm thinking of a use-case where 6 months down the line I want to audit my AWS/Vault setup with the idea of pruning old unneeded role tags.